dependency/health_service_test.go

```diff
@@ -675,6 +691,136 @@ func TestHealthServiceQuery_Fetch(t *testing.T) {
 	}
 }
 
+// TestHealthServiceQuery_Fetch_SamenessGroup tests fetching services with sameness group
+// and asserts that failover works when a service in the sameness group fails and consul-template
+// reacts to the change.
+func TestHealthServiceQuery_Fetch_SamenessGroup(t *testing.T) {
+	// Arrange - set up test data
+	catalog := testClients.Consul().Catalog()
+
+	partitionOne := "sg-partition-1"
+	partitionTwo := "sg-partition-2"
+	nodeOne := "node" + partitionOne
+	nodeTwo := "node" + partitionTwo
+	samenessGroup := "test-sameness-group"
+
+	require.NoError(t, testClients.createConsulPartition(partitionOne))
+	require.NoError(t, testClients.createConsulPartition(partitionTwo))
+	require.NoError(t, testClients.createConsulSamenessGroups(samenessGroup, partitionOne, partitionTwo))
+
+	// Register services in foo partition as critical so they will fall back to default partition service
```

Comment:
This comment is misleading I think? There's no foo partition in this test and we don't fallback to the default partition IIUC.

Reply:
fixed in edd0f7394427f64c6fc9de7aa74dd6bac9c32f80